### DISCUSSION TITLE
Why can't we put routeTree.gen to .gitignore?

### DISCUSSION THREAD

**Discussion by vipindigiqt** [TanStack/router#2902] [Q&A]
Since `routeTree.gen` gets generated automatically while running in dev and when the build gets generated (I suppose?), I thought of adding it to `.gitignore` to get it out of way but got a bunch of type errors and build was failed in my GitHub Actions pipeline.

The type errors were all this:

`Argument of type '"/some/route"' is not assignable to parameter of type 'never'.`

**Comment by osdiab** [comment]
probably because it contains a bunch of type declarations that make the typescript compile properly. If you ignore it you'd need to make sure it gets re-generated before your type checking occurs in Github Actions.

**Reply by vipindigiqt** [reply]
> probably because it contains a bunch of type declarations that make the typescript compile properly. If you ignore it you'd need to make sure it gets re-generated before your type checking occurs in Github Actions.

Yes, but there's a plugin involved here to do all of that I suppose?

We're asked to ignore the routeTree, that's not a good design decision IMHO.
